build/gen_stub.php

```diff
@@ -3046,6 +3033,40 @@ private function collectInheritedMembers(
         }
     }
 
+    /** @param array<string, ClassInfo> $classMap */
+    private function isException(array $classMap): bool
+    {
+        if ($this->name->toString() === "Throwable") {
+            return true;
+        }
+
+        foreach ($this->extends as $parentName) {
+            $parent = $classMap[$parentName->toString()] ?? null;
+            if ($parent === null) {
+                throw new Exception("Missing parent class " . $parentName->toString());
+            }
+
+            if ($parent->isException($classMap)) {
+                return true;
+            }
+        }
+
+        if ($this->type === "class") {
+            foreach ($this->implements as $interfaceName) {
+                $interface = $classMap[$interfaceName->toString()] ?? null;
+                if ($interface === null) {
+                    throw new Exception("Missing implemented interface " . $interfaceName->toString());
+                }
+
+                if ($interface->isException($classMap)) {
+                    return true;
+                }
+            }
+        }
```

Comment:
Is this just to check that ``Error`` and ``Exception`` have an ``implements Throwable``? that seems kinda overkill? logically every other exception *must* be extending one of these two classes so maybe it makes more sense to just special case those two classes and let the other loop do its job.

Either case this is also fine.

Reply:
I didn't want to rely on the fact that exceptions *currently* either extend `Error` or `Exception` so that's why I used the more general approach. I'm fine with special casing the logic if you want to.

Reply:
Eh it's fine with me :)